tests/jballerina-unit-test/src/test/resources/test-src/expressions/mappingconstructor/spread_op_field.bal

```diff
@@ -246,6 +246,37 @@ function testSpreadFieldWithRecordTypeHavingNeverField() {
     assertEquality(8, bar8.i);
 }
 
+type Vehicle record {
+    string model;
+    int year;
+};
+
+type Car record {|
+    string model;
+    int year;
+    anydata ...;
+|};
+
+type Van record {|
+    string model;
+    int year;
+    any ...;
+|};
+
+function testSpreadFieldWithClosedRecordCreatedFromOpenRecord() {
+    Vehicle vehicle1 = { model: "Tesla", year: 2023, "isElectronic": true };
+    Car car = {...vehicle1};
```

Comment:
Can we reconsider record names and structures. Generally, you'd be able to use the smaller value space type as a spread field to create a value in the larger value space.